packages/cli/test/utils/simulation/SimulationTracker.ts

```diff
@@ -279,52 +270,115 @@ export class SimulationTracker {
+    this.slotCapture.set(slot, capturedSlot);
+  }
+
+  private async processAssert({slot, epoch}: {slot: Slot; epoch: Epoch}): Promise<void> {
+    debug(`processing assert slot=${slot} epoch=${epoch}`);
     const capturedForNodes = this.slotCapture.get(slot);
     if (!capturedForNodes || capturedForNodes.length < this.nodes.length) {
       // We need to wait for all nodes to capture data for that slot
       return;
     }
+    const removeAssertions: string[] = [];
+
+    for (const node of this.nodes) {
+      for (const assertion of this.assertions) {
+        const match = assertion.match({
+          slot,
+          epoch,
+          node,
+          clock: this.clock,
+          forkConfig: this.forkConfig,
+          fork: this.forkConfig.getForkName(slot),
+        });
+
+        if (match & AssertionMatch.None || !(match & AssertionMatch.Assert)) continue;
 
-    for (const assertion of this.assertions) {
-      const match = assertion.match({slot, epoch, clock: this.clock, forkConfig: this.forkConfig});
-      if ((typeof match === "boolean" && match) || (typeof match === "object" && match.match)) {
         try {
           const errors = await assertion.assert({
+            fork: this.forkConfig.getForkName(slot),
             slot,
             epoch,
+            node,
             nodes: this.nodes,
             clock: this.clock,
             forkConfig: this.forkConfig,
-            store: this.stores[assertion.id],
-            // TODO: Make the store safe, to filter just the dependant stores not all
-            dependantStores: this.stores,
+            store: this.stores[assertion.id][node.cl.id],
+            dependantStores: getStoresForAssertions(this.stores, [assertion, ...(assertion.dependencies ?? [])]),
           });
-          if (errors) {
-            for (const err of errors) {
-              this.errors.push({slot, epoch, assertionId: assertion.id, message: err});
-            }
+
+          for (const err of errors) {
+            const message = typeof err === "string" ? err : err[0];
+            const data = typeof err === "string" ? {} : {...err[1]};
+            this.errors.push({slot, epoch, assertionId: assertion.id, nodeId: node.cl.id, message, data});
           }
         } catch (err: unknown) {
-          this.errors.push({slot, epoch, assertionId: assertion.id, message: (err as Error).message});
+          this.errors.push({
+            slot,
+            epoch,
+            nodeId: node.cl.id,
+            assertionId: assertion.id,
+            message: (err as Error).message,
+          });
         }
       }
-
-      if (typeof match === "object" && match.remove) {
-        this.removeAssertionQueue.push(assertion.id);
-      }
     }
 
-    this.reporter.progress(slot);
-    this.processRemoveAssertionQueue();
-  }
-
-  private processRemoveAssertionQueue(): void {
-    for (const id of this.removeAssertionQueue) {
+    for (const id of removeAssertions) {
```

Comment:
@nazarhussain Noticed that removeAssertions is always empty here as there are no items pushed to array after initializing it

https://github.com/ChainSafe/lodestar/blob/29314c9e4515662fd882b9a47b8aa0c95b0ff88c/packages/cli/test/utils/simulation/SimulationTracker.ts#L330